src/modules/authentacation/components/ChangePassword/ChangePassword.tsx

```diff
@@ -1,7 +1,129 @@
-import React from 'react'
+import React, { useState } from 'react'
+import { useForm } from 'react-hook-form'
+import { PASSWORD_VALIDATION } from '../../../../services/validation/validation'
+import { axiosInstance, USERS_URLS } from '../../../../services/apisUrls/apisUrls'
+import { useNavigate } from 'react-router-dom'
+import { toast } from 'react-toastify'
+import { FaRegEyeSlash, FaSpinner } from 'react-icons/fa'
+import { IoEyeOutline } from 'react-icons/io5'
+
+interface formData {
+  oldPassword: string,
+  newPassword: string,
+  confirmNewPassword: string
+}
 
 export default function ChangePassword() {
+  const [isPasswordVisable, setIsPasswordVisable] = useState(false);
+  const [isOldPasswordVisable, setIsOldPasswordVisable] = useState(false);
+  const [isConfirmPasswordVisable, setIsConfirmPasswordVisable] = useState(false);
+
+  const navigate = useNavigate()
+  const { register, formState: { errors, isSubmitting }, handleSubmit }
+    = useForm<formData>()
+
+  const onSubmit = async (data: any) => {
+    try {
+      let response = await axiosInstance.put(USERS_URLS.CHANGE_PASSWORD, data)
+      console.log(response);
+      navigate("/login")
+      toast.success("password updated succeffully")
+    } catch (error) {
+      console.log(error);
+      toast.error(response.data.error)
```

Comment:
I suggest using optional chaining to avoid runtime errors if the backend changes the response structure.